models/components.go

```diff
@@ -97,40 +89,100 @@ func (component Component) GetAsEnvironment() map[string]interface{} {
 	}
 }
 
-func (c *Component) Summarize() Summary {
-	if c.Summary.processed {
+func (c *Component) Summarize() types.Summary {
+	if c.Summary.IsProcessed() {
 		return c.Summary
 	}
 
-	var s Summary
+	var s types.Summary
 	s.Incidents = c.Summary.Incidents
 	s.Insights = c.Summary.Insights
 
 	if c.Components == nil {
 		switch c.Status {
-		case ComponentStatusHealthy:
+		case types.ComponentStatusHealthy:
 			s.Healthy++
-		case ComponentStatusUnhealthy:
+		case types.ComponentStatusUnhealthy:
 			s.Unhealthy++
-		case ComponentStatusWarning:
+		case types.ComponentStatusWarning:
 			s.Warning++
-		case ComponentStatusInfo:
+		case types.ComponentStatusInfo:
 			s.Info++
 		}
 		s.Incidents = c.Summary.Incidents
 		s.Insights = c.Summary.Insights
-		s.processed = true
+		s.SetProcessed(true)
 		return s
 	}
 
 	for _, child := range c.Components {
 		childSummary := child.Summarize()
 		s = s.Add(childSummary, c.Name+"-"+child.Name)
 	}
-	s.processed = true
+	s.SetProcessed(true)
 	return s
 }
 
+func (component Component) GetID() string {
+	if component.ID != uuid.Nil {
+		return component.ID.String()
+	}
+	if component.Text != "" {
+		return component.Text
+	}
+	return component.Name
+}
```

Comment:
Is this function used anywhere ? Ideally it should always return id and throw error on it being nil

Reply:
I see that it's only used in the `Debug` method. https://github.com/flanksource/duty/blob/4fd7266159f67e96ff1cbd179a5ef9a7daf5cbe0/models/components.go#L207

Reply:
Ohh, we can delete this method then